docs/antrea-network-policy.md

```diff
@@ -540,6 +588,98 @@ The [ovs-pipeline doc](design/ovs-pipeline.md) contains more information on how
 policy rules are realized by OpenFlow, and how the priority of flows reflects the
 order in which they are enforced.
 
+## ClusterGroup
+
+A ClusterGroup (CG) CRD is a specification of how workloads are grouped together.
+It allows admins to group Pods using traditional label selectors, which can then
+be referenced in ACNP in place of stand alone `podSelector` and/or `namespaceSelector`.
+ClusterGroups allow admins to separate the concern of grouping of workloads from
+the security aspect of Antrea-native policies.
+It adds another level of indirection allowing users to update group membership
+without having to update individual policy rules.
+In addition to specifying label selectors to group workloads, admins can create a
+ClusterGroup to share IPBlocks.
+An `ipBlock` selector may not be specified with a `podSelector` and `namespaceSelector`,
+i.e. a single ClusterGroup can either group workloads or share IPBlocks.
+A ClusterGroup is cluster scoped resource and therefore can only be set in an Antrea
+ClusterNetworkPolicy's `appliedTo` and `to`/`from` peers.
+
+### The ClusterGroup resource
+
+An example ClusterGroup might look like this:
+
+```yaml
+apiVersion: core.antrea.tanzu.vmware.com/v1alpha2
+kind: ClusterGroup
+metadata:
+  name: test-cg-sel
+spec:
+  podSelector:
+    matchLabels:
+      role: db
+  namespaceSelector:
+    matchLabels:
+      env: prod
+status:
+  conditions:
+    - type: "GroupMembersComputed"
+      status: "True"
+      lastTransitionTime: "2021-01-29T19:59:39Z"
+---
+apiVersion: core.antrea.tanzu.vmware.com/v1alpha2
+kind: ClusterGroup
+metadata:
+  name: test-cg-ip-block
+spec:
+  # IPBlock cannot be set along with PodSelector, NamespaceSelector.
+  ipBlock:
+    cidr: 10.0.10.0/24
+status:
+  conditions:
+    - type: "GroupMembersComputed"
+      status: "True"
+      lastTransitionTime: "2021-01-29T19:59:39Z"
+---
+```
+
+**spec**: The ClusterGroup `spec` has all the information needed to define a
+cluster-wide group.
+
+**podSelector**: Pods can be grouped cluster-wide using `podSelector`.
+If set with a `namespaceSelector`, all matching Pods from Namespaces selected
+by the `namespaceSelector` will be grouped.
+
+**namespaceSelector**: All Pods from Namespaces selected by the namespaceSelector
+will be grouped.
+If set with a `podSelector`, all matching Pods from Namespaces selected by the
+`namespaceSelector` will be grouped.
+
+**ipBlock**: This selects a particular IP CIDR range to allow as `ingress`
+"sources" or `egress` "destinations".
+A ClusterGroup with `ipBlock` set may not be used in an ACNP's `appliedTo` field.
```

Comment:
Maybe repeat what you said in the earlier section as well: "There will be no error, but the policy will have no effect"

Reply:
done